kernel_tuner/util.py

```diff
@@ -570,6 +570,24 @@ def get_total_timings(results, env, overhead_time):
     return env
 
 
+def to_valid_nvrtc_gpu_arch_cc(compute_capability: str) -> str:
+    """Returns a valid Compute Capability for NVRTC `--gpu-architecture=`, as per https://docs.nvidia.com/cuda/nvrtc/index.html#group__options."""
+    valid_cc = ['50', '52', '53', '60', '61', '62', '70', '72', '75', '80', '87', '89', '90', '90a']    # must be in ascending order, when updating also update test_to_valid_nvrtc_gpu_arch_cc
+    compute_capability = str(compute_capability)
+    if len(compute_capability) < 2:
+        raise ValueError(f"Compute capability '{compute_capability}' must be at least of length 2, is {len(compute_capability)}")
+    if compute_capability in valid_cc:
+        return compute_capability
+    # if the compute capability does not match, scale down to the nearest matching
+    subset_cc = [cc for cc in valid_cc if compute_capability[0] == cc[0]]
+    if len(subset_cc) > 0:
+        # get the next-highest valid CC
+        highest_cc_index = max([i for i, cc in enumerate(subset_cc) if int(cc[1]) <= int(compute_capability[1])])
+        return subset_cc[highest_cc_index]
```

Comment:
This seems like a rather complicated way of trying to match the input `compute_capability` to something in `valid_cc`. I am sure there must be a way to make it simpler and more readable.

Reply:
I agree, the problem is we can't use integer comparison because it's also possible to have e.g. `90a`, and we can only match within the major number. Any suggestions?

Reply:
How about using difflib, e.g. the [`get_close_matches` ](https://docs.python.org/3/library/difflib.html#difflib.get_close_matches?) function.